File: 22.10_PS/1022_소수_and_팰린드롬.java
// boj. 1747

import java.io.*;
import java.util.*;


public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int MAX_VALUE = (int)1e7;
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[MAX_VALUE + 1];
        arr[0] = 1;
        arr[1] = 1;

        for (int i = 0; i <= Math.sqrt(MAX_VALUE); i++) {
            if (arr[i] == 0) {
                int num = 2;
                while (i * num <= MAX_VALUE) {
                    arr[i * num] = 1;
                    num ++;
                }
            }
        }

        for (int j = n; j <= MAX_VALUE; j++) {
            if (arr[j] == 0 && isPalindrome(j)) {
                System.out.println(j);
                break;
            }
        }

    }

    static boolean isPalindrome(int number) {
        String param = Integer.toString(number);
        String[] pArr = param.split("");
        int idx = 0;
        while (idx <= pArr.length / 2) {
            if (!pArr[idx].equals(pArr[pArr.length - 1 - idx]))
                return false;
            idx++;
        }
        return true;
    }
}

File: 22.10_PS/1022_소수_and_팰린드롬.py
# boj. 1747

import sys

def is_palindrome(num):
    length = len(str(num))
    div_num = pow(10, length - 1)
    for _ in range(length // 2):
        a = num // div_num
        b = num % 10
    
        if (a != b):
            return False
        num = (num % div_num) // 10
        div_num //= 100
    return True


input = sys.stdin.readline

INF = int(1e7) 
n = int(input())
flag = False # 수의 범위가 n을 넘었는지 체크

arr = [True] * (INF + 1)
arr[0], arr[1] = False, False

# 에라토스테네스의 체
for i in range(2, 1000 + 1):
    if arr[i] == True:
        num = 2
        while i * num <= INF:
            arr[i * num] = False
            num += 1
    
for i in range(n, INF + 1):
    if arr[i] == True and is_palindrome(i):
        sys.exit(print(i))

